pkg/client/unversioned/transport.go

```diff
@@ -171,30 +171,42 @@ func NewClientCertTLSConfig(certData, keyData, caData []byte) (*tls.Config, erro
 	if err != nil {
 		return nil, err
 	}
-	certPool := x509.NewCertPool()
-	certPool.AppendCertsFromPEM(caData)
+
 	return &tls.Config{
 		// Change default from SSLv3 to TLSv1.0 (because of POODLE vulnerability)
 		MinVersion: tls.VersionTLS10,
 		Certificates: []tls.Certificate{
 			cert,
 		},
-		RootCAs:    certPool,
-		ClientCAs:  certPool,
+		RootCAs:    rootCertPool(caData),
 		ClientAuth: tls.RequireAndVerifyClientCert,
```

Comment:
Can you drop ClientAuth as well (that's another server only option), and update downstream to make sure all tests pass?